src/features/references/editor/ReferenceDetailsCard.tsx

```diff
@@ -0,0 +1,261 @@
+const EditableTableDataCell = ({
+  contentData,
+  id,
+  onChangeHandler,
+}: {
+  contentData: string;
+  id: string;
+  onChangeHandler: CallableFunction;
+}) => (
+  <td className="w-auto p-5">
+    <DataTextInput content={contentData} id={id} onChangeHandler={onChangeHandler} />
+  </td>
+);
+
+const UnEditableTableDataCell = ({ contentData, id }: { contentData: string | Author[]; id: string }) => {
+  const contentString = getContentString(contentData, id);
+
+  return (
+    <td className="w-auto p-5">
+      <span className="leading-[30px]">{contentString}</span>
+    </td>
+  );
+};
+
+const TableHeadCell = ({ content, colSpan, header }: { content: string; colSpan?: number; header?: boolean }) => {
+  const colSpanString = { ['colSpan']: colSpan };
+
+  return (
+    <th
+      className={cx(
+        { 'rounded-t bg-card-bg-header p-3 text-white': header },
+        { 'w-1/6 p-6': !header },
+        'text-sm uppercase',
+      )}
+      key={content}
+      {...colSpanString}
+    >
+      {content}
+      {IconButtons(header)}
+    </th>
+  );
+};
+
+const IconButtons = (header: boolean | undefined): ReactElement[] => {
+  if (header) {
+    return [
+      IconButton({ hiddenWhenEditing: true, icon: EditIcon(), title: 'Edit Reference' }),
+      IconButton({ hiddenWhenEditing: false, icon: CloseIcon(), title: 'Finished Editing Reference' }),
+    ];
+  } else {
+    return [];
+  }
+};
+
+const IconButton = ({
+  hiddenWhenEditing,
+  title,
+  icon,
+}: {
+  hiddenWhenEditing: boolean;
+  title: string;
+  icon: ReactElement;
+}) => {
+  const { editing, setEditing } = useContext(EditingContext);
+  const hiddenClass = hiddenWhenEditing ? { hidden: editing } : { hidden: !editing };
+
+  return (
+    <button
+      className={cx(hiddenClass, 'float-right hover:text-cyan-100')}
+      key={title}
+      title={title}
+      onClick={() => {
+        setEditing(!editing);
+      }}
+    >
+      {icon}
+    </button>
+  );
+};
+
+export default function ReferenceDetailsCard({
+  reference,
+  handleReferenceChanged,
+}: {
+  reference: ReferenceItem;
+  handleReferenceChanged: (params: ReferenceItem) => void;
+}) {
+  const [editing, setEditing] = useState(false);
+  const cloneReferenceItem = <T extends object>(source: T): T => ({
+    ...source,
+  });
+
+  const updatedReference = cloneReferenceItem(reference);
+
+  // We have to use a copy of the reference for updating purposes, otherwise UpdateReferencesAtom sees no changes and doesn't update
+  const referenceChanged = useCallback(
+    (fieldName: string, value: string) => {
+      switch (fieldName) {
+        case 'citationKey':
+          updatedReference.citationKey = value;
+          break;
+        case 'title':
+          updatedReference.title = value;
+          break;
+        case 'doi':
+          updatedReference.doi = value;
+          break;
+      }
+      handleReferenceChanged(updatedReference);
+    },
+    [updatedReference, handleReferenceChanged],
+  );
```

Comment:
Could you create `updatedReference` inside `referenceChanged`: no need to copy the reference on every render, you can do it only when we actually need to update it.
Also nit: the convention for such a function name is usually something like `handleReferenceChange`